pkgs/top-level/all-packages.nix

```diff
@@ -12541,6 +12541,13 @@ with pkgs;
     stdenv = overrideCC stdenv gcc6;
   };
 
+  webkitgtk222x = callPackage ../development/libraries/webkitgtk/2.22.nix {
+    harfbuzz = harfbuzzFull;
+    inherit (gst_all_1) gst-plugins-base gst-plugins-bad;
+    stdenv = overrideCC stdenv gcc6;
```

Comment:
have you tried using a newer GCC?

Reply:
I tested this and it seems we are running in some issues:

```
[2835/2838] Generating ../../WebKit2-4.0.gir
FAILED: WebKit2-4.0.gir
cd /build/webkitgtk-2.22.2/build/Source/WebKit && CC=/nix/store/7qv7drgc2jadd5r0xi3mfvvf570v2r7s-gcc-wrapper-7.3.0/bin/gcc CFLAGS=-Wno-deprecated-declarations LDFLAGS= LD_LIBRARY_PATH="/build/webkitgtk-2.22.2/build/lib" /nix/store/33n761k36gjvhs1pza2zgscnzicd6ss9-gobject-introspection-1.56.0-dev/bin/g-ir-scanner --quiet --warn-all --symbol-prefix=webkit --identifier-prefix=WebKit --namespace=WebKit2 --nsversion=4.0 --include=GObject-2.0 --include=Gtk-3.0 --include=Soup-2.4 --include-uninstalled=/build/webkitgtk-2.22.2/build/JavaScriptCore-4.0.gir --library=webkit2gtk-4.0 --library=javascriptcoregtk-4.0 -L/build/webkitgtk-2.22.2/build/lib --no-libtool --pkg=gobject-2.0 --pkg=gtk+-3.0 --pkg=libsoup-2.4 --pkg-export=webkit2gtk-4.0 --output=/build/webkitgtk-2.22.2/build/WebKit2-4.0.gir --c-include="webkit2/webkit2.h" -DBUILDING_WEBKIT -DWEBKIT2_COMPILATION -I/build/webkitgtk-2.22.2/Source -I/build/webkitgtk-2.22.2/Source/WebKit -I/build/webkitgtk-2.22.2/build/DerivedSources -I/build/webkitgtk-2.22.2/build/DerivedSources/webkit2gtk -I/build/webkitgtk-2.22.2/build/DerivedSources/JavaScriptCore/javascriptcoregtk -I/build/webkitgtk-2.22.2/build/DerivedSources/ForwardingHeaders -I/build/webkitgtk-2.22.2/build/DerivedSources/ForwardingHeaders/JavaScriptCore/glib -I/build/webkitgtk-2.22.2/build/DerivedSources/ForwardingHeaders/webkit2gtk /build/webkitgtk-2.22.2/build/DerivedSources/webkit2gtk/webkit2/WebKitEnumTypes.h /build/webkitgtk-2.22.2/build/DerivedSources/webkit2gtk/webkit2/WebKitVersion.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitApplicationInfo.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitAuthenticationRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitAutocleanups.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitAutomationSession.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitBackForwardList.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitBackForwardListItem.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitCredential.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitContextMenu.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitContextMenuActions.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitContextMenuItem.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitCookieManager.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitDefines.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitDownload.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitEditingCommands.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitEditorState.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitError.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitFaviconDatabase.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitFileChooserRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitFindController.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitFormSubmissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitForwardDeclarations.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitGeolocationPermissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitHitTestResult.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitInstallMissingMediaPluginsPermissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitJavascriptResult.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitMimeInfo.h
/build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitNavigationAction.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitNavigationPolicyDecision.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitNetworkProxySettings.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitNotificationPermissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitNotification.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitOptionMenu.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitOptionMenuItem.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitPermissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitPlugin.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitPolicyDecision.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitResponsePolicyDecision.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitScriptDialog.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitSecurityManager.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitSecurityOrigin.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitSettings.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitURIRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitURIResponse.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitURISchemeRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitUserContent.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitUserContentManager.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitUserMediaPermissionRequest.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebContext.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebResource.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebView.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebViewSessionState.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebsiteData.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWebsiteDataManager.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/WebKitWindowProperties.h /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/webkit2.h /build/webkitgtk-2.22.2/Source/WebKit/Shared/API/glib/*.cpp /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/glib/*.cpp /build/webkitgtk-2.22.2/Source/WebKit/UIProcess/API/gtk/*.cpp
/nix/store/dxf1m7dhc4qb655bdljc1fsd74v1nag3-binutils-2.30/bin/ld: /build/webkitgtk-2.22.2/build/Source/WebKit/tmp-introspectbU9AxO/WebKit2-4.0: hidden symbol `__divmodti4' in /nix/store/s4rr4gsn08wdj2pq7dmia91ksm3xw602-gcc-7.3.0/lib/gcc/x86_64-unknown-linux-gnu/7.3.0/libgcc.a(_divmoddi4.o) is referenced by DSO
/nix/store/dxf1m7dhc4qb655bdljc1fsd74v1nag3-binutils-2.30/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
linking of temporary binary failed: Command '['/nix/store/7qv7drgc2jadd5r0xi3mfvvf570v2r7s-gcc-wrapper-7.3.0/bin/gcc', u'-o', u'/build/webkitgtk-2.22.2/build/Source/WebKit/tmp-introspectbU9AxO/WebKit2-4.0', '-Wno-deprecated-declarations', u'/build/webkitgtk-2.22.2/build/Source/WebKit/tmp-introspectbU9AxO/WebKit2-4.0.o', '-L.', '-Wl,-rpath,.', '-Wl,--no-as-needed', '-lwebkit2gtk-4.0', '-ljavascriptcoregtk-4.0', '-L/build/webkitgtk-2.22.2/build/lib', '-Wl,-rpath,/build/webkitgtk-2.22.2/build/lib', u'-L/nix/store/nndjz8l466v2i8rjcc339qalf77gichr-glib-2.56.0/lib', u'-L/nix/store/nhxjjkiwvprca9jh2jy6jh53x1z27ql4-libsoup-2.62.2/lib', u'-L/nix/store/9l8bdcrh88lwcdb2x1i8s0q31gj9ld06-gtk+3-3.22.30/lib', u'-L/nix/store/pcpa1dwfpfd8ycdc060mkqwi9xwyrivx-cairo-1.15.12/lib', u'-L/nix/store/xl9w0xg22d0sfbajjcn12r08clkyfc8d-pango-1.42.4/lib', u'-L/nix/store/jqpwvr6jwpag1qnq5c7bhphdwdaldair-gdk-pixbuf-2.36.12/lib', u'-L/nix/store/qh2zlizxka16z4wa03h4aj81i04iwjhd-atk-2.28.1/lib', u'-Wl,--export-dynamic', u'-lgmodule-2.0', u'-pthread', u'-lgtk-3', u'-lgdk-3', u'-lpangocairo-1.0', u'-lpango-1.0', u'-latk-1.0', u'-lcairo-gobject', u'-lcairo', u'-lgdk_pixbuf-2.0', u'-lsoup-2.4', u'-lgio-2.0', u'-lgobject-2.0', u'-lglib-2.0']' returned non-zero exit status 1
ninja: build stopped: subcommand failed.
builder for '/nix/store/m7m1x00mg0dmsg59rfqsm7fbsd80dfwf-webkitgtk-2.22.2.drv' failed with exit code 1
cannot build derivation '/nix/store/dkrgp1q030njhqcw4qy2zlmk8rgz3j4p-astroid-0.14.drv': 1 dependencies couldn't be built
error: build of '/nix/store/dkrgp1q030njhqcw4qy2zlmk8rgz3j4p-astroid-0.14.drv' failed
```

I wouldn't deal with this right now. gcc6 should be good enough since that is also what upstream switched to in ~April.

Reply:
I tried after you asked and yes, it fails on my machine, too